apps/nextjs-app/src/features/app/blocks/space/space-side-bar/SpaceItem.tsx

```diff
@@ -14,26 +19,85 @@ interface IProps {
 export const SpaceItem: React.FC<IProps> = ({ space, isActive }) => {
   const { id, name } = space;
   const ref = useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const queryClient = useQueryClient();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { mutateAsync: updateSpaceMutator } = useMutation({
+    mutationFn: updateSpace,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ReactQueryKeys.spaceList() });
+      queryClient.invalidateQueries({ queryKey: ReactQueryKeys.space(space.id) });
+    },
+  });
+
+  useEffect(() => {
+    if (isEditing) setTimeout(() => inputRef.current?.focus());
+  }, [isEditing]);
+
+  useClickAway(inputRef, async () => {
+    if (isEditing && inputRef.current?.value && inputRef.current.value !== space.name)
```

Comment:
better to enclose the if statement
```suggestion
    if (isEditing && inputRef.current?.value && inputRef.current.value !== space.name) {
    await updateSpaceMutator({
        spaceId: space.id,
        updateSpaceRo: { name: inputRef.current.value },
      });
}
```